### PR TITLE
slice: clean up 502 header/body generation.

### PR DESCRIPTION
**Pull request by traeak**
Small change in how corner case 502 is generated.
Also change the header debug to use TSHttpHdrPrint and pull the string off the TSIOBuffer blocks instead of building it through iteration.

### CODE REVIEW

File: plugins/experimental/slice/slice.h [hunk 1]
```diff
@@ -32,9 +32,11 @@
 #define PLUGIN_NAME "slice"
 #endif
 
+/*
```
Comment (shinrich) [inline plugins/experimental/slice/slice.h:35]: Maybe just take this away if you don't want it?

File: plugins/experimental/slice/response.cc [hunk 2]
```diff
@@ -68,10 +68,13 @@ string502()
     bodystr.append("</body>\n");
     bodystr.append("</html>\n");
 
+    char hverstr[64];
+    int const hlen =
+      snprintf(hverstr, sizeof(hverstr), "HTTP/%d.%d 502 Bad Gateway\r\n", TS_HTTP_MAJOR(httpver), TS_HTTP_MINOR(httpver));
+    msg.append(hverstr, hlen);
+
```
Comment (shinrich) [inline plugins/experimental/slice/response.cc:75]: I assume this is the heart of your 502 fixes for this PR.  Much of the rest looks like more general cleanup.

Comment (traeak) [reply plugins/experimental/slice/response.cc:75]: Yes sort of.  The original PR was much more aggressive about trying to use the state machine but attempts failed and this cleanup was what was left over.

**Comment by traeak** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
Backported to 9.0.1 in #7435

**Pull request merged**